DQMOffline/Hcal/src/HcalRecHitsAnalyzer.cc

```diff
@@ -60,28 +60,54 @@ HcalRecHitsAnalyzer::HcalRecHitsAnalyzer(edm::ParameterSet const& conf) {
 
     es.get<CaloGeometryRecord > ().get(geometry);
 
-    const std::vector<DetId>& hbCells = geometry->getValidDetIds(DetId::Hcal, HcalBarrel);
-    const std::vector<DetId>& heCells = geometry->getValidDetIds(DetId::Hcal, HcalEndcap);
-    const std::vector<DetId>& hoCells = geometry->getValidDetIds(DetId::Hcal, HcalOuter);
-    const std::vector<DetId>& hfCells = geometry->getValidDetIds(DetId::Hcal, HcalForward);
-
-    nChannels_[1] = hbCells.size(); 
-    nChannels_[2] = heCells.size(); 
-    nChannels_[3] = hoCells.size(); 
-    nChannels_[4] = hfCells.size();
+    const CaloGeometry* geo = geometry.product();
```

Comment:
Hi David, geometry is also used in the analyze method.

I've changed HcalRecHitsDQMClient so that it is no longer a member variable.